### PR TITLE
Simple servlet to search Apache Solr

### LINKED ISSUE #1: Create new middleware layer to interface with Solr
### 💡 Description

Per IT security concerns, we need to create a simple passthrough server that will provide an interface between the general public and the Apache Solr Legacy Registry.

### ⚔️ Parent Epic / Related Tickets

_No response_

### PR DESCRIPTION
**Pull request by jordanpadams**
## 🗒️ Summary
Simple servlet to sit between keyword search / public access and Solr.

## ⚙️ Test Data and/or Report
```
# Build package
% mvn package

# Move package to Apache Tomcat webapps dir
% mv target/*.war /path/to/tomcat/webapps

# Test some queries from http://localhost:8080/registry-legacy-server/search
```

Some test queries:
* [http://localhost:8080/registry-legacy-server/search?q=*:*](http://localhost:8080/registry-legacy-server/search?q=*:*)
  * then select any of the available facets to see them work

## ♻️ Related Issues

Resolves NASA-PDS/search-ui-legacy#5

### CODE REVIEW

File: src/changes/changes.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  may be required before exporting such information to foreign countries or
+  providing access to foreign nationals.
+
+  $Id$
+-->
+
+<document>
+  <properties>
+    <title>Release Changes</title>
+    <author email="Sean.Hardman@jpl.nasa.gov">Sean Hardman</author>
```
Comment (nutjob4life) [inline src/changes/changes.xml:22]: Really? Sean Hardman? 😁

File: src/main/assembly/tar-assembly.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+
+<!-- 
+  Copyright 2014, by the California Institute of Technology.
```
Comment (nutjob4life) [inline src/main/assembly/tar-assembly.xml:4]: 2024 😉

File: src/main/java/gov/nasa/pds/search/servlet/RegistryLegacyServlet.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+              this.solrRequestHandler, queryString);
+      LOG.info("URL: " + url);
+
+
+      HttpClient client = HttpClient.newHttpClient();
+      HttpRequest solrRequest = HttpRequest.newBuilder().uri(URI.create(url)).build();
+
+
+      HttpResponse<String> solrResponse = client.send(solrRequest, BodyHandlers.ofString());
+      // LOG.info(String.valueOf(solrResponse.statusCode()));
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/search/servlet/RegistryLegacyServlet.java:129]: Maybe this is just me, but I don't like to see commented-out code unless there's an explanation of why it's commented-out. Better to add that or just delete it!

File: src/main/java/gov/nasa/pds/search/servlet/SolrSearch.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  public SolrSearch(String url) {
+    this.solrServerUrl = url;
+    LOG.info("Solr Server URL: " + url);
+  }
+
+  /**
+   * Search method for querying the Solr Service and returning a list
+   * of Solr Documents.
+   */
+//  public SolrDocumentList search(String query, int start, int rows) throws MalformedURLException, SolrServerException {
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/search/servlet/SolrSearch.java:47]: Can we just delete all this? See earlier remark about commented-out code. It should be explained or deleted—with prejudice 😈

**Review by nutjob4life — CHANGES_REQUESTED** [review]
I couldn't actually get this to run with Tomcat 10, 9, 8, or 7 (via Docker) but maybe that's a "me problem" 😆

Otherwise the servlet implementation looks fine. I'd just like to see the commented-out code explained or deleted 👍

**Comment by jordanpadams** [comment]
@nutjob4life thanks for the review. moved to https://github.com/NASA-PDS/search-ui-legacy/pull/6 to bring some of these components together to simplify the deployment and since they are so tightly coupled.

**Pull request closed without merging**